x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/aggs/InferencePipelineAggregationBuilder.java

```
@@ -0,0 +1,236 @@
+                                                            String pipelineAggregatorName,
+                                                            XContentParser parser) {
+        Tuple<SetOnce<ModelLoadingService>, String> context = new Tuple<>(modelLoadingService, pipelineAggregatorName);
+        return PARSER.apply(parser, context);
+    }
+
+    public InferencePipelineAggregationBuilder(String name, SetOnce<ModelLoadingService> modelLoadingService,
+                                               Map<String, String> bucketsPath) {
+        super(name, NAME, new TreeMap<>(bucketsPath).values().toArray(new String[] {}));
+        this.modelLoadingService = modelLoadingService;
+        this.bucketPathMap = bucketsPath;
+    }
+
+    public InferencePipelineAggregationBuilder(StreamInput in, SetOnce<ModelLoadingService> modelLoadingService) throws IOException {
+        super(in, NAME);
+        modelId = in.readString();
+        bucketPathMap = in.readMap(StreamInput::readString, StreamInput::readString);
+        inferenceConfig = in.readOptionalNamedWriteable(InferenceConfigUpdate.class);
+        gapPolicy = BucketHelpers.GapPolicy.readFrom(in);
+        this.modelLoadingService = modelLoadingService;
+    }
+
+    void setModelId(String modelId) {
+        this.modelId = modelId;
+    }
+
+    void setInferenceConfig(InferenceConfigUpdate inferenceConfig) {
+        this.inferenceConfig = inferenceConfig;
+    }
+
+    void setGapPolicy(BucketHelpers.GapPolicy gapPolicy) {
+        this.gapPolicy = gapPolicy;
+    }
+
+    @Override
+    protected void validate(ValidationContext context) {
+        context.validateHasParent(NAME, name);
+        if (modelId == null) {
+            context.addValidationError("Model Id must be set");
+        }
+        if (gapPolicy != BucketHelpers.GapPolicy.SKIP) {
+            context.addValidationError("gap policy [" + gapPolicy + "] in not valid for [" + NAME + "] aggregation");
```

Since it only accepts `skip`, you could just remove the option from the parser and hardcode it.

---

GapPolicy is a common setting, it feels a little underhand to silently ignore the setting and rewrite it to `skip`.

---

I think we'd just make it not something the parser recognized at all. Lots of aggs don't have a gap policy.

---

Yep, I meant removing it from the parser, so it'd throw an exception if the user tried to set it.  Most pipeline aggs have a gap_policy but it's not a required feature of pipelines by any means.

---

👍  I assumed pipeline aggs must accept a gap_policy but that doesn't make sense in every case. 

The code was silly anyway I was parsing the field validating it then never reading the setting again. It's gone now.